test/integration/test_account_client.py

```diff
@@ -116,37 +109,6 @@ def test_custom_client_app_header(self):
                 custom_header, client._session.headers["X-Qx-Client-Application"]
             )
 
-    def test_access_token_not_in_exception_traceback(self):
-        """Check that access token is replaced within chained request exceptions."""
-        backend_name = "ibmq_qasm_simulator"
-        backend = self.provider.get_backend(
-            backend_name, hub=self.hub, group=self.group, project=self.project
-        )
-        circuit = transpile(self.qc1, backend, seed_transpiler=self.seed)
-        qobj = assemble(circuit, backend, shots=1)
-        client = backend._api_client
-
-        exception_message = (
-            "The access token in this exception "
-            "message should be replaced: {}".format(self.access_token)
-        )
-        exception_traceback_str = ""
-        try:
-            with mock.patch.object(
-                HTTPConnectionPool,
-                "urlopen",
-                side_effect=MaxRetryError(
-                    HTTPConnectionPool("host"), "url", reason=exception_message
-                ),
-            ):
-                _ = client.job_submit(backend.name(), qobj.to_dict())
-        except RequestsApiError:
-            exception_traceback_str = traceback.format_exc()
-
-        self.assertTrue(exception_traceback_str)
-        if self.access_token in exception_traceback_str:
-            self.fail("Access token not replaced in request exception traceback.")
-
```

Comment:
We no longer store the `access_token` as an instance attribute on the session. Instead we use the built-in authentication approach (https://github.com/Qiskit/qiskit-ibm-provider/blob/1428b86dc724fa4f2e3cae9253e561b07df3ee72/qiskit_ibm_provider/api/auth.py#L21). That way `requests` will be smart enough to not include the access token in the stack trace and there's no need to manually strip it off.